### DISCUSSION TITLE
Pyinstaller skipping imported submodule

### DISCUSSION THREAD

**Discussion by caffeinepills** [pyinstaller/pyinstaller#6525] [Help] [answered]
I have just upgraded to the latest Pyinstaller and Python 3.10 from Python 3.6. I have verified my script runs with the new Python version. My script previously was able to be converted into an exe without issues prior to this.

However, I am having an issue with an important submodule being skipped over during the analysis. I have verified it's not present in the Analysis-00 file. This is not a site package, it's a module that is part of my script.

For example there are two modules/files:
`client.scene.login`
`client.interface.login`

At the top of `client.scene.login` it has `import client.interface.login`, but is not found during analysis. The `client.scene.login` *is* found.

I have tried adding  `hiddenimports=['client.interface.login']` to my spec file but no change.

My project is too large to post, but maybe someone can lead me in the right direction on how I would go about debugging why it's being skipped over, or forcing it to be included during analysis?

I don't see any errors during the pyinstaller process. I have also verified that I get `client.interface.login` cannot be found when running the produced exe.

EDIT:
I found this in the log warn log:
`invalid module named client.interface.login - imported by client.scene.login (top-level)`
Xref file labels it as : `InvalidSourceModule`
How can I find out why it was labeled as invalid? Is there a way to enable a more verbose debug? 

EDIT2: 
Found and resolved it.

I went into the code of `pyinstaller/lib/modulegraph/modulegraph.py` and found this block

```py
                except SyntaxError:
                    co = None
                    cls = InvalidSourceModule
```

And change the exception to print the error, which pointed me in the right direction. 
TLDR: An old function that was no longer in use, but analysis found it which caused the module to fail inspection. Would be nice if it showed a message like the generic `except Exception` below it.

**Comment by caffeinepills** [comment, marked as answer]
Resolved.
